src/installer/corehost/cli/fxr/fx_muxer.cpp

```diff
@@ -870,8 +870,27 @@ int fx_muxer_t::run_app(host_context_t *context)
 
 int fx_muxer_t::get_runtime_delegate(host_context_t *context, coreclr_delegate_type type, void **delegate)
 {
-    if (context->is_app)
-        return StatusCode::InvalidArgFailure;
+    switch (type)
+    {
+    case coreclr_delegate_type::com_activation:
+    case coreclr_delegate_type::load_in_memory_assembly:
+    case coreclr_delegate_type::winrt_activation:
+    case coreclr_delegate_type::com_register:
+    case coreclr_delegate_type::com_unregister:
+        if (context->is_app)
+            return StatusCode::HostApiUnsupportedScenario;
+        break;
+    default:
+        // Always allowed
+        break;
+    }
+
+    if (type > coreclr_delegate_type::load_assembly_and_get_function_pointer
```

Comment:
Nit: could you add a short comment (similar to what you have in hostpolicy.cpp) pointing out how we can't just always use the `last_known_delegate_type` and the significance of `load_assembly_and_get_function_pointer` as the 'original' last value.